pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -348,3 +348,13 @@ def _check_on_configure_sharded_model(trainer: "pl.Trainer") -> None:
                 "The `on_configure_sharded_model` callback hook was deprecated in"
                 " v1.6 and will be removed in v1.8. Use `setup()` instead."
             )
+
+def _check_on_pretrain_routine_start_end(trainer: "pl.Trainer") -> None:
```

Comment:
Rebase and add this logic to the function that checks all the deprecated callback hooks instead of iterating through the full callback list again

Reply:
Can you please share a sample code to only check for deprecated callbacks?

Reply:
See the function here: https://github.com/PyTorchLightning/pytorch-lightning/blob/8394770d4afa5480f881229b150ac44eaa8c41b0/pytorch_lightning/trainer/configuration_validator.py#L313